pkg/cmd/server/origin/controller/hpa.go

```diff
@@ -0,0 +1,49 @@
+package controller
+
+import (
+	hpacontroller "k8s.io/kubernetes/pkg/controller/podautoscaler"
+	hpametrics "k8s.io/kubernetes/pkg/controller/podautoscaler/metrics"
+
+	osclient "github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+)
+
+// NB: this is funky -- it's actually a Kubernetes controller, but we run it as an OpenShift controller in order
+// to get a handle on OpenShift clients, so that our delegating scales getter can work.
+
+type HorizontalPodAutoscalerControllerConfig struct {
+	HeapsterNamespace string
+}
+
+func (c *HorizontalPodAutoscalerControllerConfig) RunController(ctx ControllerContext) (bool, error) {
+	hpaClient := ctx.KubeControllerContext.ClientBuilder.ClientOrDie(bootstrappolicy.InfraHorizontalPodAutoscalerControllerServiceAccountName)
+
+	// use the Kubernetes config so that the service account is in the same name namespace for both clients
+	hpaOriginConfig := ctx.KubeControllerContext.ClientBuilder.ConfigOrDie(bootstrappolicy.InfraHorizontalPodAutoscalerControllerServiceAccountName)
```

Comment:
heh.  I almost did that in the last round of revisions, but figured it wasn't worth the extra `if err return err` lines for what should be identical behavior.  I'll add it in.